tests/test_helpers.py

```diff
@@ -975,3 +975,59 @@ def test_populate_with_cookies():
 )
 def test_parse_http_date(value, expected):
     assert parse_http_date(value) == expected
+
+
+@pytest.mark.parametrize(
+    ["netrc_contents", "hostname", "expected_username"],
+    [
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            "username",
+        ),
+    ],
+    indirect=("netrc_contents",),
+)
+@pytest.mark.usefixtures("netrc_contents")
+def test_netrc_from_env(netrc_contents: str, hostname: str, expected_username: str):
+    """Test that reading netrc files from env works as expected"""
+    netrc_obj = helpers.netrc_from_env()
+    assert netrc_obj.authenticators(hostname)[0] == expected_username
+
+
+@pytest.mark.parametrize(
+    ["netrc_contents", "hostname", "expected_auth"],
+    [
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            helpers.BasicAuth("username", "pass"),
+        ),
+        (
+            "machine example.com account username password pass\n",
+            "example.com",
+            helpers.BasicAuth("username", "pass"),
+        ),
+        (
+            "machine example.com password pass\n",
+            "example.com",
+            helpers.BasicAuth("", "pass"),
+        ),
+        ("", "example.com", None),
+    ],
+    indirect=("netrc_contents",),
+)
+@pytest.mark.usefixtures("netrc_contents")
+def test_basicauth_from_netrc(
+    netrc_contents: str,
```

Comment:
This is incorrect: `netrc_contents` currently returns `NoneType` and not `str`. But with my suggestion above, it'll be `pathlib.Path`.

OTOH, since you're not using it in the test function, you don't need to list it as an arg at all — `@pytest.mark.usefixtures("netrc_contents")` takes care of the injection. But if you end up using it, you won't need `@pytest.mark.usefixtures("netrc_contents")`.